src/frequenz/sdk/_internal/_channels.py

```diff
@@ -0,0 +1,26 @@
+# License: MIT
+# Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+"""General purpose classes for use with channels."""
+
+import abc
+import typing
+
+from frequenz.channels import Receiver
+
+T = typing.TypeVar("T")
+
+
+class ReceiverFetcher(typing.Generic[T], typing.Protocol):
+    """An interface that just exposes a `new_receiver` method."""
+
+    @abc.abstractmethod
+    def new_receiver(self, maxsize: int = 50) -> Receiver[T]:
+        """Get a receiver from the channel.
+
+        Args:
+            maxsize: The maximum size of the receiver.
+
+        Returns:
+            A receiver instance.
+        """
```

Comment:
It was made to provide a single type around `MetricAggregator` and the `ChannelRegistry`.  But you're right, the internal channel registry type doesn't have to derive from `ReceiverFetcher`.  Done here: https://github.com/frequenz-floss/frequenz-sdk-python/pull/692/commits/50fc8d34062b64df7bbb9b6e39f6a983fb7e5dfa

Reply:
OK, my point was that probably in this case it made more sense to just not use a `Protocol` and derive both instead, but I guess that's also more likely to end with circular dependencies (I guess that's another nice feature of `Protocol`). I'm still not convinced which is better to be honest, but since I'm not convinced I will leave the topic here.

Almost. If we start using `Protocol` more, actually `ReceiverFetcher` (maybe with another name) could be moved to `channels`, as most channels are also `ReceiverFetcher`s. Also I think `Sender` and `Receiver` could be protocols.

Reply:
I agree about moving it to the channels repo, already have a todo item for it,  because there's really no nice place to put it in the sdk, especially because it has to be publicly exposed, and it is a bit weird now, living in `_internal`, but still exposed publicly.